src/REveMu2eMainWindow.cc

```diff
@@ -252,84 +150,259 @@ void REveMu2eMainWindow::GeomDrawer(TGeoNode* node, REX::REveTrans& trans, REX::
+        x_pt = offsets[i].second[0];
+        y_pt  =offsets[i].second[1];
+        z_pt = offsets[i].second[2];
+      }
+      }
+      std::vector<double> shift;
+      shift.push_back(-1*x_sol);
+      shift.push_back(-1*y_sol);
+      shift.push_back(z_sol);
+      // set tracker to be central in the frame (shift to 0,0,0)
+      if(geomOpt.showPS){
+        static std::vector <std::string> substring_ps  {"PSVacuum"};
+        for(auto& i: substring_ps){
+          showNodesByName(node,i,kFALSE, 0, trans, beamlineholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("BLColor"));
+        }
+        static std::vector <std::string> substring_pt  {"ProductionTarget"};
+        for(auto& i: substring_pt){
+          std::vector<double> shiftpt;
+          shiftpt.push_back(x_pt);
+          shiftpt.push_back(y_pt);
+          shiftpt.push_back(z_pt);
+          showNodesByName(node,i,kFALSE, 0, trans, beamlineholder, maxlevel, level, false, false, shiftpt, false, true, drawconfigf.getInt("BLColor"));
+        }
+      }
+      if(geomOpt.showTS){
+        static std::vector <std::string> substring_ts  {"TS1Vacuum","TS1CryoInsVac","TS2Vacuum","TS2CryoInsVac","TS3Vacuum","TS3CryoInsVac","TS4Vacuum","TS4CryoInsVac","TS5Vacuum","TS5CryoInsVac"};
+        for(auto& i: substring_ts){
+          showNodesByName(node,i,kFALSE, 0, trans, beamlineholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("BLColor"));
+        }
+      }
+      if(geomOpt.showDS){
+        static std::vector <std::string> substring_ds  {"DS*"};
+        for(auto& i: substring_ds){
+          showNodesByName(node,i,kFALSE, 0, trans, beamlineholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("BLColor"));
+        }
+      }
+  }
+
+void REveMu2eMainWindow::GeomDrawerNominal(TGeoNode* node, REX::REveTrans& trans, REX::REveElement* beamlineholder, REX::REveElement* trackerholder, REX::REveElement* caloholder, REX::REveElement* crystalsholder, REX::REveElement* crvholder, REX::REveElement* targetholder, int maxlevel, int level, GeomOptions geomOpt, std::vector<std::pair<std::string, std::vector<float>>>& offsets){
+
+    for(unsigned int i = 0; i < offsets.size(); i++){
+      if(offsets[i].first.find("World") != string::npos){
+        x_world = offsets[i].second[0];
+        y_world = offsets[i].second[1];
+        z_world = offsets[i].second[2];
+      }
+      if(offsets[i].first.find("HallAir") != string::npos)
+      {
+        x_hall = x_world + offsets[i].second[0];
+        y_hall = y_world + offsets[i].second[1];
+        z_hall = z_world + offsets[i].second[2];
+      }
+      if(offsets[i].first.find("DS3Vacuum") != string::npos)      {
+        x_ds3 = x_hall + offsets[i].second[0];
+        y_ds3 = y_hall + offsets[i].second[1];
+        z_ds3 = z_hall + offsets[i].second[2];
+        z_crv = -1*offsets[i].second[2];//595.3
```

Comment:
These comments  may not age well.  If there is a good reason to keep them, name a git tag or SHA for which they are correct.